systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/KafkaRollerST.java

```diff
@@ -322,6 +328,107 @@ void testKafkaPodPendingDueToRack(ExtensionContext extensionContext) {
         KafkaUtils.waitForKafkaDeletion(namespaceName, clusterName);
     }
 
+    /**
+     * @description This test case verifies the rolling update behavior of Kafka controller nodes under specific conditions.
+     * It focuses on ensuring that changes in Kafka configuration affect only the intended node pools, particularly
+     * the controller nodes, while leaving others like broker nodes unaffected.
+     *
+     * @steps
+     *  1. - Assume that KRaft mode is enabled and the installation method is bundle only.
+     *  2. - Initialize the test environment and create a Kafka instance with a single replica.
+     *  3. - Create and deploy a Kafka node pool with broker roles (brokerPool) and another with controller roles (controllerPool), each with 3 replicas.
+     *  4. - Take snapshots of the broker and controller pods for later comparison.
+     *  5. - Update a specific Kafka configuration that affects only controller nodes and verify the rolling update behavior.
+     *     - Ensure that only controller nodes undergo a rolling update, while broker nodes remain unaffected.
+     *  6. - Verify the rolling update of controller nodes by comparing the snapshots taken before and after the configuration change.
+     *
+     * @usecase
+     *  - kafka-controller-node-rolling-update
+     *  - kafka-configuration-change-impact
+     *  - kafka-node-pool-management
+     */
+    @ParallelNamespaceTest
+    void testKafkaRollingUpdatesOfControllerNodes(final ExtensionContext extensionContext) {
```

Comment:
Just an idea ... this is good. But could we add a second rolling update related to some change in the controller node pool (e.g. some affinity, pod label, health check or something like that not related to Kafka configuration)? That will help to trigger also slightly different code path.

Reply:
Okay, I extend the test case with another RollinUpdate with PodAffinity :) Thanks for this comment.